src/Persistence/Sql.php

```diff
@@ -631,14 +607,14 @@ public function action(Model $m, $type, $args = [])
                 break;
             case 'count':
                 $this->initQueryConditions($m, $q);
-                $m->hook(self::HOOK_INIT_SELECT_QUERY, [$q]);
-                if (isset($args['alias'])) {
-                    $q->reset('field')->field('count(*)', $args['alias']);
-                } else {
-                    $q->reset('field')->field('count(*)');
-                }
+                $m->hook(self::HOOK_INIT_SELECT_QUERY, [$q, $type]);
 
-                return $q;
+                return $q->reset('field')->field('count(*)', $args['alias'] ?? null);
+            case 'exists':
+                $this->initQueryConditions($m, $q);
```

Comment:
refactor m to model and q to query

Reply:
I am for it but let's refactor this after the merge. Otherwise the PR is contaminated.